src/analysis/mellin_transform.lean

```diff
@@ -0,0 +1,333 @@
+      refine (integrable.has_finite_integral _).const_mul _,
+      rw [←integrable_on, ←integrable_on_Ioc_iff_integrable_on_Ioo,
+        ←interval_integrable_iff_integrable_Ioc_of_le hε.le],
+      exact interval_integral.interval_integrable_rpow' (by linarith) },
+    { refine (ae_restrict_iff' measurable_set_Ioo).mpr (eventually_of_forall $ λ t ht, _),
+      rw [mul_comm, norm_mul],
+      specialize hε' _ ht.1,
+      { rw [dist_eq_norm, sub_zero, norm_of_nonneg (le_of_lt ht.1)],
+        exact ht.2 },
+      { refine (mul_le_mul_of_nonneg_right hε' (norm_nonneg _)).trans _,
+        simp_rw [norm_of_nonneg (rpow_nonneg_of_nonneg (le_of_lt ht.1) _), mul_assoc],
+        refine mul_le_mul_of_nonneg_left (le_of_eq _) hd.le,
+        rw ←rpow_add ht.1,
+        congr' 1,
+        abel } } },
+end
+
+/-- If `f` is a locally integrable real-valued function on `Ioi 0` which is `O(x ^ (-a))` at `∞`
+and `O(x ^ (-b))` at `0`, then its Mellin transform integral converges for `b < s < a`. -/
+lemma mellin_convergent_of_is_O_scalar
+  {a b : ℝ} {f : ℝ → ℝ} {s : ℝ}
+  (hfc : locally_integrable_on f $ Ioi 0)
+  (hf_top : is_O at_top f (λ t, t ^ (-a))) (hs_top : s < a)
+  (hf_bot : is_O (𝓝[Ioi 0] 0) f (λ t, t ^ (-b))) (hs_bot : b < s) :
+  integrable_on (λ t : ℝ, t ^ (s - 1) * f t) (Ioi 0) :=
+begin
+  obtain ⟨c1, hc1, hc1'⟩ := mellin_convergent_top_of_is_O hfc.ae_strongly_measurable hf_top hs_top,
+  obtain ⟨c2, hc2, hc2'⟩ := mellin_convergent_zero_of_is_O hfc.ae_strongly_measurable hf_bot hs_bot,
+  have : Ioi 0 = Ioc 0 c2 ∪ Ioc c2 c1 ∪ Ioi c1,
+  { rw [union_assoc, Ioc_union_Ioi (le_max_right _ _), Ioc_union_Ioi
+    ((min_le_left _ _).trans (le_max_right _ _)), min_eq_left (lt_min hc2 hc1).le] },
+  rw [this, integrable_on_union, integrable_on_union],
+  refine ⟨⟨hc2', integrable_on_Icc_iff_integrable_on_Ioc.mp _⟩, hc1'⟩,
+  refine (hfc.continuous_on_mul _ is_open_Ioi).integrable_on_compact_subset
+    (λ t ht, (hc2.trans_le ht.1 : 0 < t)) is_compact_Icc,
+  exact continuous_at.continuous_on (λ t ht, continuous_at_rpow_const _ _ $ or.inl $ ne_of_gt ht),
+end
+
+lemma mellin_convergent_of_is_O_rpow [normed_space ℂ E]
+  {a b : ℝ} {f : ℝ → E} {s : ℂ}
+  (hfc : locally_integrable_on f $ Ioi 0)
+  (hf_top : is_O at_top f (λ t, t ^ (-a))) (hs_top : s.re < a)
+  (hf_bot : is_O (𝓝[Ioi 0] 0) f (λ t, t ^ (-b))) (hs_bot : b < s.re) :
+  integrable_on (λ t : ℝ, (t : ℂ) ^ (s - 1) • f t) (Ioi 0) :=
+begin
+  rw mellin_convergent_iff_norm (subset_refl _) measurable_set_Ioi
+    hfc.ae_strongly_measurable,
+  exact mellin_convergent_of_is_O_scalar
+    hfc.norm hf_top.norm_left hs_top hf_bot.norm_left hs_bot,
+end
+
+end mellin_convergent
+
+section mellin_diff
+
+/-- If `f` is `O(x ^ (-a))` at `+∞`, then `log • f` is `O(x ^ (-b))` for every `b < a`. -/
+lemma is_O_rpow_top_log_smul [normed_space ℝ E] {a b : ℝ} {f : ℝ → E}
+  (hab : b < a) (hf : is_O at_top f (λ t, t ^ (-a))) :
+  is_O at_top (λ t : ℝ, t.log • f t) (λ t, t ^ (-b)) :=
```

Comment:
```suggestion
  is_O at_top (λ t : ℝ, log t • f t) (λ t, t ^ (-b)) :=
```
We try to follow the usual conventions when writing the usual functions. If the problem is between real and complex logarithm, you can use `hiding log` when you `open complex`. Same thing below several times.

Reply:
OK, fixed! The problem was indeed distinguishing between `real.log` and `complex.log`, so I've followed your suggestion of `import complex hiding log`.